ts/Dashboards/Components/Component.ts

```diff
@@ -828,6 +827,35 @@ abstract class Component {
 
     }
 
+    /**
+     * Fixes the sync options to be in the correct format.
+     *
+     * @internal
+     */
+    protected fixSyncOptions(): void {
```

Comment:
It took me a while to figure out what you are trying to `fix` here.
Wouldn't the `convertToSyncOptions` or simply `setSyncOptions`  be better?